runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,314 @@
+		graphics.fillRect(x, y, WIDTH, height);
+
+		// draw bar background
+		graphics.setColor(DEPLETED);
+		graphics.fillRect(x + PADDING, y + PADDING, WIDTH - PADDING * OFFSET, height - PADDING * OFFSET);
+
+		// draw bar with current health or prayer points
+		int filledHeight = getBarHeight(max, current, height);
+		graphics.setColor(filled);
+		graphics.fillRect(x + PADDING, y + PADDING + (height - filledHeight), WIDTH - PADDING * OFFSET, filledHeight - PADDING * OFFSET);
+	}
+
+	private void renderHealingBar(Graphics2D graphics, int x, int y, int max, int current, int height, int heal, Color color, Image image, String counterText, int counterX)
+	{
+		final int OVERHEAL_OFFSET = 2;
+		final int HEAL_OFFSET = 3;
+		final Color OVERHEAL_COLOR = new Color(216, 255, 139, 150);
+
+		// Start drawing the healing bar
+		int filledHeight = getBarHeight(max, heal, height);
+		int filledCurrentHeight = getBarHeight(max, current, height);
+		int overHeal;
+		graphics.setColor(color);
+
+		if (filledHeight + filledCurrentHeight > height)
+		{
+			overHeal = filledHeight + filledCurrentHeight - height;
+			filledHeight = filledHeight - overHeal + OVERHEAL_OFFSET;
+			color = OVERHEAL_COLOR;
+			graphics.setColor(color);
+			graphics.fillRect(x + PADDING, y - filledCurrentHeight + (height - filledHeight) + HEAL_OFFSET,
+					WIDTH - PADDING * OVERHEAL_OFFSET, filledHeight - PADDING * OVERHEAL_OFFSET);
+		}
+		else
+		{
+			graphics.fillRect(x + PADDING, y - OVERHEAL_OFFSET - filledCurrentHeight + (height - filledHeight) + HEAL_OFFSET,
+					WIDTH - PADDING * OVERHEAL_OFFSET, filledHeight + OVERHEAL_OFFSET - PADDING * OVERHEAL_OFFSET);
+		}
+
+		final int ICON_AND_COUNTER_OFFSET_X = 1;
+		final int ICON_AND_COUNTER_OFFSET_Y = 21;
+		final int SKILL_ICON_HEIGHT = 35;
+		final int COUNTER_ICON_HEIGHT = 18;
+
+		if (config.enableCounter() || config.enableSkillIcon())
+		{
+			// draw counter on top of everything else
+			graphics.setFont(FontManager.getRunescapeSmallFont());
+			textComponent.setColor(Color.WHITE);
+
+			// If the prayer has skill icons disabled then assign noImage
+			if (config.enableSkillIcon() && config.enableCounter())
+			{
+				//draw icons on top of everything else
+				graphics.drawImage(image, x + ICON_AND_COUNTER_OFFSET_X + PADDING, y + ICON_AND_COUNTER_OFFSET_Y - image.getWidth(null), null);
+				textComponent.setPosition(new java.awt.Point(x + counterX + PADDING, y + SKILL_ICON_HEIGHT));
+				textComponent.setText(counterText);
+				textComponent.render(graphics);
+			}
+			else if (config.enableCounter() && !config.enableSkillIcon())
+			{
+				textComponent.setPosition(new java.awt.Point(x + counterX + PADDING, y + COUNTER_ICON_HEIGHT));
+				textComponent.setText(counterText);
+				textComponent.render(graphics);
+			}
+			else if (!config.enableCounter() && config.enableSkillIcon())
+			{
+				graphics.drawImage(image, x + ICON_AND_COUNTER_OFFSET_X + PADDING, y + ICON_AND_COUNTER_OFFSET_Y - image.getWidth(null), null);
+			}
```

Comment:
This block can be reduced down to two if statements where one does the imageDrawing and the other does the text drawing.